File: 프로그래머스/0/120000/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
class Solution {
    public int solution(String my_string) {
        int answer = 0;
        StringBuilder str_num = new StringBuilder();
        for (char ch : my_string.toCharArray()) {
            if (Character.isDigit(ch)) {
                str_num.append(ch);
            } else {
                if (str_num.length() > 0) {
                    answer += Integer.parseInt(str_num.toString());
                    str_num.setLength(0);
                }
            }
        }
        if (str_num.length() > 0) {
            answer += Integer.parseInt(str_num.toString());
        }
        return answer;
    }
}

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.py
def solution(my_string):
    answer = 0
    str = ''
    for i in my_string:
        if i.isdigit():
            str += i
        else :
            if str:
                answer += int(str)
                str = ''
    if str:
        answer += int(str)
    return answer